### PR TITLE
Support Floats

### PR DESCRIPTION
**Pull request by ryancco**
When trying to convert a floating point number, the $value parameter would be converted to an integer. This will support floating point numbers to be converted without being rounded to the nearest full integer.

### CODE REVIEW

**Comment by ash-jc-allen** [comment]
Hey @ryancco! Thanks for your pull request. Would you be able to give an example of when/why you'd need to use a float instead on an int? 😄

**Comment by ryancco** [comment]
> Hey @ryancco! Thanks for your pull request. Would you be able to give an example of when/why you'd need to use a float instead on an int? 😄

Hey, sure thing! When converting, for example, $10.52 from one currency to another this would previously coerce 10.52 to the closest integer and convert that amount. When working with monetary values that contain non round amounts as I do in a few apps, this can (and, unfortunately, has been) resulting in sporadic amounts sometimes resulting in loss of funds for ourselves and/or the clients using the platforms. 

tl;dr it doesn’t support converting monetary amounts with change.

**Comment by ash-jc-allen** [comment]
I have a feeling you might be using the package slightly wrong. All the conversions in this package are done using ints instead of floats for these kinds of reasons. So in your example where you'd be using `10.52`, you should be passing it in as `1052`. This is because you need to pass it in using the currency's lowest denomination. This should stop you from getting the rounding errors that you're getting.

There's some notes about it in the docs and some examples if that helps? 😄

https://github.com/ash-jc-allen/laravel-exchange-rates#convert-currencies

**Comment by ryancco** [comment]
I understand this is a best practice, although not sure how it would avoid *these* issues but rather enforce that good practice. I guess that's something I can get behind though nonetheless. Thanks, I'll close this in that case :)

**Comment by ash-jc-allen** [comment]
No problem. Thanks for reaching out though. If you're still having any issues with it further down the line, give me a shout :)
